spec/spec_helper.rb

```diff
@@ -22,21 +23,52 @@ def establish_database_connection
   ActiveRecord::Base.establish_connection(
     adapter: "postgresql",
     host: ENV.fetch("PGHOST", "localhost"),
-    user: ENV.fetch("PGUSER", "postgres"),
-    password: ENV.fetch("PGPASSWORD", ""),
+    user: ENV.fetch("PGUSER", "ubuntu"),
+    password: ENV.fetch("PGPASSWORD", "password"),
     database: ENV.fetch("PGDATABASE", "que-test"),
   )
 end
 
 establish_database_connection
 
+
+class LockDatabaseRecord < ActiveRecord::Base
+  establish_connection(
+      adapter: "postgresql",
+      host: ENV.fetch("LOCK_PGHOST", "localhost"),
+      user: ENV.fetch("LOCK_PGUSER", "ubuntu"),
+      password: ENV.fetch("LOCK_PGPASSWORD", "password"),
+      database: ENV.fetch("LOCK_PGDATABASE", "lock-test"),
+      port: ENV.fetch("LOCK_PGPORT", 5435),
+      pool: 6,)
+end
+
+class YugabyteRecord < ActiveRecord::Base
+    establish_connection(
+      adapter: "postgresql",
+      host: ENV.fetch("PGHOST", "localhost"),
+      user: ENV.fetch("PGUSER", "ubuntu"),
+      password: ENV.fetch("PGPASSWORD", "password"),
+      database: ENV.fetch("PGDATABASE", "que-test"),
+    )
+end
+
 # Make sure our test database is prepared to run Que
-Que.connection = ActiveRecord
+if ENV['YUGABYTE_QUE_WORKER_ENABLED']
```

Comment:
Not sure if you've started productionising the code yet, but I'd perhaps put all of this extra testing config in another file since this is an extension, rather than read an env var.

Remember that the specs are the first user of the code, so if it is awkward to configure then we need to rethink the interface